### PR TITLE
Enabled commutative property for MatrixElement

### PR DESCRIPTION
**Pull request by arihantparsoya**
This is fix for #11077 

Previously the `is_commutative` property was `None` and hence Sympy wasnt able to simplify the determinant expression.
- After setting the `is_commutative` property to `True`, the determinant expression is being simplified.
- None of the tests fail after changing the `is_commutative` property

### CODE REVIEW

**Comment by parsoyaarihant** [comment]
@asmeurer , can you review this?

**Comment by asmeurer** [comment]
Looks like some tests failed. Those should be investigated.

File: sympy/matrices/expressions/matexpr.py [hunk 1]
```diff
@@ -338,7 +338,7 @@ class MatrixElement(Expr):
     i = property(lambda self: self.args[1])
     j = property(lambda self: self.args[2])
     _diff_wrt = True
-
+    is_commutative = True
```
Comment (asmeurer) [inline sympy/matrices/expressions/matexpr.py]: Keep a blank line before function definitions.

**Comment by asmeurer** [comment]
You should add a test for the original issue.

**Comment by parsoyaarihant** [comment]
There are in total 4 test failures, one is related to the PR [#9364](https://github.com/sympy/sympy/pull/9364) and the remaining 3 are related to printing which I fixed in my last commit. I will look into [#9364](https://github.com/sympy/sympy/pull/9364) and fix the last remining test case.

**Comment by jksuom** [comment]
I wonder if there could be a way of making matrix elements commutative on demand. Matrices with elements in non-commutative rings are generally quite acceptable. Commutativity is needed only in some special cases such as for determinants.

**Comment by parsoyaarihant** [comment]
@asmeurer , should I add the tests for the determinant of the matrix (i.e the issue for determinant which u originally posted) or just the commutability of the MatrixElement?

**Comment by parsoyaarihant** [comment]
@asmeurer , can you review this?

File: sympy/polys/tests/test_polytools.py [hunk 2]
```diff
@@ -2881,7 +2882,7 @@ def test_cancel():
     M = MatrixSymbol('M', 5, 5)
     assert cancel(M[0,0] + 7) == M[0,0] + 7
     expr = sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2] / z
-    assert cancel(expr) == expr
+    assert simplify(cancel(expr)) == expr
```
Comment (asmeurer) [inline sympy/polys/tests/test_polytools.py]: This test seems wrong. The point of the function is to test `cancel`. What does this do without `simplify` here?

**Comment by asmeurer** [comment]
I would add a test for the commutativity (which you already have), and a test for the determinant, to make sure that that works.

File: sympy/polys/tests/test_polytools.py (hunk 2 above)
Comment (asmeurer) [inline sympy/polys/tests/test_polytools.py]: I guess it should return `(z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z`. If that's what it's giving, we should test that.

Comment (parsoyaarihant) [inline sympy/polys/tests/test_polytools.py]: yeah it returns `cancel(expr) = (z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z` and previously it was returning the expression `sin(M[1, 4] + 5*M[2, 1]*M[4, 0]) - 5*M[1, 2]/z`, please note that the whole numerator is encloses in a bracket after `is_commutative` is set to `True` hence I used `simplify`

**Comment by parsoyaarihant** [comment]
@asmeurer , I added the test for determinant of the matrix.

File: sympy/polys/tests/test_polytools.py (hunk 2 above)
Comment (asmeurer) [inline sympy/polys/tests/test_polytools.py]: `(z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z` is the right answer for cancel to return. The test should be

```py
assert cancel(expr) == (z*sin(M[1, 4] + M[2, 1] * 5 * M[4, 0]) - 5 * M[1, 2]) / z
```

Calling `simplify` in the function that tests the output of `cancel` defeats the purpose.

**Comment by asmeurer** [comment]
Just the comment on the `cancel` test. The rest looks good.

**Comment by parsoyaarihant** [comment]
@asmeurer , I fixed the test.

**Comment by parsoyaarihant** [comment]
@asmeurer , can you review this again?

**Comment by asmeurer** [comment]
Looks good, thanks.

**Comment by asmeurer** [comment]
Oh, one thing. In the future, if you fix an issue, put "Fixes #11077" in one of your commit messages. That way, when the PR is merged, the issue will be automatically closed.

**Pull request merged**